frontend/src/pages/BYONImages/BYONImagesTable.tsx

```diff
@@ -292,130 +273,113 @@ export const BYONImagesTable: React.FC<BYONImagesTableProps> = ({ images, forceU
           setUpdateImageModalVisible(false);
         }}
       />
-      <Toolbar data-id="toolbar-items">
-        <ToolbarContent>{items}</ToolbarContent>
-      </Toolbar>
-      <TableComposable
+      <Table
         className={tableFilter.count === 0 ? 'empty-table' : ''}
         aria-label="Notebook images table"
         variant="compact"
-      >
-        <Thead>
+        data={images}
+        toolbarContent={items}
+        columns={columns}
+        emptyTableView={
           <Tr>
-            <Th />
-            <Th sort={getSortParams(0)}>{columnNames.name}</Th>
-            <Th sort={getSortParams(1)}>{columnNames.description}</Th>
-            <Th>Enable</Th>
-            <Th sort={getSortParams(4)}>{columnNames.user}</Th>
-            <Th sort={getSortParams(5)}>{columnNames.uploaded}</Th>
-            <Th />
+            <Td colSpan={8}>
```

Comment:
I wonder if we can infer the Tr and Td colspan code based on columns.length. That way we remove this boilerplate code from everyone.

Reply:
It usually is (as i assume this is for empty states). However some empty states have additional logic or are formatted differntly so they are pulled out of the component. 

```tsx
<Tr>
              <Td colSpan={mappedColumns.length} style={{ textAlign: 'center' }}>
                No projects match your filters.{' '}
                <Button variant="link" isInline onClick={clearFilters}>
                  Clear filters
                </Button>
              </Td>
            </Tr>
```

Like this old implementation uses the columns length. colSpan is only ever used in deprecated components (ProjectTable), outdated components (BYON), or inside a sub component. 

we could add a way to add colSpan inside the columns list, but inferring could get confusing to people imo